File: 백준/Silver/11723. 집합/집합.py
input = __import__('sys').stdin.readline

S = set()

M = int(input())
for i in range(M):
    arr = list(input().split())
    if arr[0] == "add":
        S.add(int(arr[1]))
    elif arr[0] == "remove":
        S.discard(int(arr[1]))
    elif arr[0] == "check":
        if int(arr[1]) in S:
            print(1)
        else:
            print(0)
    elif arr[0] == "toggle":
        if int(arr[1]) in S:
            S.remove(int(arr[1]))
        else:
            S.add(int(arr[1]))
    elif arr[0] == "all":
        S = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20}
    elif arr[0] == "empty":
        S = set()

File: 백준/Silver/11723. 집합/집합.cc
#include <iostream>
#include <string>
using namespace std;

int main() {
    ios_base :: sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int arr[30] = {};

    int M;
    cin >> M;

    string str;
    int x;
    for (int i=0; i<M; i++) {
        cin >> str;
        if (str == "add") {
            cin >> x;
            arr[x] = 1;

        } else if (str == "remove") {
            cin >> x;
            arr[x] = 0;

        } else if (str == "check") {
            cin >> x;
            cout << (arr[x] == 1 ? 1 : 0) << '\n';

        } else if (str == "toggle") {
            cin >> x;
            if (arr[x] == 1) arr[x] = 0;
            else arr[x] = 1;

        } else if (str == "all") {
            for (int j=1; j<=20; j++) arr[j] = 1;

        } else if (str == "empty") {
            for (int j=1; j<=20; j++) arr[j] = 0;

        }
    }
    return 0;
}